src/Middleware/ResponseCaching/src/ResponseCachingMiddleware.cs

```diff
@@ -157,6 +161,18 @@ internal async Task<bool> TryServeCachedResponseAsync(ResponseCachingContext con
                 {
                     _logger.NotModifiedServed();
                     context.HttpContext.Response.StatusCode = StatusCodes.Status304NotModified;
+
+                    if (context.CachedResponseHeaders != null)
+                    {
+                        for (var i = 0; i < HeadersToIncludeIn304.Length; i++)
```

Comment:
```suggestion
                        for (var key in HeadersToIncludeIn304)
```

Reply:
you mean `foreach` I guess. I thought a `for` loop was more performant here but it looks like I was wrong.

Reply:
`foreach` and `for` should be comparable when the type you're iterating over is "concrete" rather than an interface. For example `foreach`ing over a `string[]` or `List<string>` is *fast*, but for `IList<string>`, `IEnumerable<string>` using `for` is faster. Interface-dispatch makes the different ;).